app/policies/order/delivery_status_policy.rb

```diff
@@ -0,0 +1,28 @@
+class Order::DeliveryStatusPolicy
+
+  def initialize(user, order)
+    @user = user
+    @order = order
+  end
+
+  def mark_delivered?
+    return false if user.buyer_only?
+    return false unless order.undelivered_for_user?(user)
+    user.can_manage_market?(order.market) || delivery_schedule_is_direct_to_customer?
+  end
+
+  def mark_undelivered?
+    (order.market.is_buysell_market? && !order.undelivered_for_user?(user) && user.can_manage_market?(order.market)) ||
+      (order.market.is_consignment_market? && user.can_manage_market?(order.market) &&
+        (order.delivery_status == 'delivered' || order.delivery_status == 'partially delivered') && Inventory::Utils.consignment_can_undeliver?(order))
```

Comment:
TODO: Review delegate/refactor

Reply:
@thermistor I'm thinking since consignment code will get moved into new branch soon it might be okay leaving it as is? If you're okay with it, I'm thinking to punt on multiple policies here, but I defer to you.

Reply:
Sure. I had an example written yesterday but forgot to share. I would just extract the main || into an if-else. Something like:
``` 
if order.market.is_buysell_market?
  # other buysell conditions
else
  # consignment conditions
end
```